hypothesis-python/tests/cover/test_stateful.py

```diff
@@ -352,6 +353,43 @@ def execute_step(self, step):
         assert False
 
 
+class MachineWithConsumingRule(RuleBasedStateMachine):
+    b1 = Bundle("b1")
+    b2 = Bundle("b2")
+
+    def __init__(self):
+        self.created_counter = 0
+        self.consumed_counter = 0
+        super(MachineWithConsumingRule, self).__init__()
+
+    @invariant()
+    def bundle_length(self):
+        assert len(self.bundle("b1")) == self.created_counter - self.consumed_counter
+
+    @rule(target=b1)
+    def populate_b1(self):
+        self.created_counter += 1
+        return self.created_counter
+
+    @rule(target=b2, consumed=consumes(b1))
+    def depopulate_b1(self, consumed):
+        self.consumed_counter += 1
+        return consumed
+
+    @rule(value1=b1, value2=b2)
+    def check(self, value1, value2):
+        assert value1 != value2
+
+
+def test_consumes():
```

Comment:
Idiom: nstead of this function, we usually just assign `TestMachineWithConsumingRule = MachineWithConsumingRule.TestCase()` and let pytest discover the resulting class.

I'd also love to see a test or rule that uses collections of values, e.g. `lists(consumes(b1))`.